src/metrics/rules/active_ruleset.go

```diff
@@ -619,29 +654,24 @@ func (as *activeRuleSet) cutoverNanosAt(idx int) int64 {
 	return timeNanosMax
 }
 
-// mergeResultsForExistingID merges the next staged metadata into the current list of staged
-// metadatas while ensuring the cutover times of the staged metadatas are non-decreasing. This
+// ensureNonDecreasingCutoverForExistingID ensures the most recently added staged metadata in the
+// current list of staged metadatas has a cutover time greater than or equal to the one before it. This
 // is needed because the cutover times of staged metadata results produced by mapping rule matching
 // may not always be in ascending order. For example, if at time T0 a metric matches against a
 // mapping rule, and the filter of such rule changed at T1 such that the metric no longer matches
 // the rule, this would indicate the staged metadata at T0 would have a cutover time of T0,
 // whereas the staged metadata at T1 would have a cutover time of 0 (due to no rule match),
 // in which case we need to set the cutover time of the staged metadata at T1 to T1 to ensure
 // the mononicity of cutover times.
-func mergeResultsForExistingID(
-	currMetadatas metadata.StagedMetadatas,
-	nextMetadata metadata.StagedMetadata,
+func ensureNonDecreasingCutoverForExistingID(
```

Comment:
This method is repurposed to keep cutover times accurate since the metadata itself has already been merged correctly into the MatchResult